seatunnel-connectors-v2/connector-file/connector-file-ftp/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/ftp/util/FtpFileUtils.java

```diff
@@ -0,0 +1,179 @@
+                        FTPCLIENT.setConnectTimeout(FTP_CONNECT_MAX_TIMEOUT);
+                        FTPCLIENT.setControlEncoding("utf-8");
+                        FTPCLIENT.enterLocalPassiveMode();
+                        FTPCLIENT.setFileType(FTP.BINARY_FILE_TYPE);
+                        if (!FTPReply.isPositiveCompletion(FTPCLIENT.getReplyCode())) {
+                            FTPCLIENT.disconnect();
+                        }
+                    } catch (SocketException e) {
+                        e.printStackTrace();
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                }
+            }
+        }
+        return FTPCLIENT;
+    }
+
+    public static OutputStream getOutputStream(@NonNull String outFilePath) throws IOException {
+        FTPClient ftpClient = FtpFileUtils.getFTPClient();
+        if (!fileExist(outFilePath)) {
+            createFile(outFilePath);
+        }
+        OutputStream outputStream = ftpClient.appendFileStream(new String(outFilePath.getBytes("UTF-8"), "iso-8859-1"));
+        return outputStream;
+    }
+
+    public static boolean createDir(@NonNull String dirPath) throws IOException {
+        FTPClient ftpClient = getFTPClient();
+        StringTokenizer s = new StringTokenizer(dirPath, "/"); // sign
+        s.countTokens();
+        String pathName = "";
+        while (s.hasMoreElements()) {
+            pathName = pathName + "/" +  s.nextElement();
+            try {
+                ftpClient.makeDirectory(pathName);
+            } catch (Exception e) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public static boolean createFile(@NonNull String filePath) throws IOException {
+        FTPClient ftpClient = getFTPClient();
+        StringTokenizer s = new StringTokenizer(filePath, "/"); // sign
+        int c = s.countTokens();
+        String dirName = "";
+        for (int i = 0; i < c - 1; i++) {
+            dirName = dirName + "/" +  s.nextElement();
+        }
+        createDir(dirName);
+        byte[] b = "".getBytes();
+        InputStream inputStream = new ByteArrayInputStream(b);
+        ftpClient.storeFile(filePath, inputStream);
+        return true;
+    }
+
+    public static boolean fileExist(@NonNull String filePath) throws IOException {
+        FTPClient ftpClient = getFTPClient();
+        String[] listNames = ftpClient.listNames(filePath);
+        if (listNames != null) {
+            return listNames.length > 0;
+        }
+        else {
+            return false;
+        }
+    }
+
+    public static void renameFile(@NonNull String oldName, @NonNull String newName) throws IOException {
+        LOGGER.info("begin rename file oldName :[" + oldName + "] to newName :[" + newName + "]");
+        FTPClient ftpClient = getFTPClient();
+        if (!FtpFileUtils.fileExist(newName)) {
+            FtpFileUtils.createFile(newName);
```

Comment:
hi, why did we need to create?

Reply:
I think this step is judging the target directory is or not existed. Because before transaction commit the files are existed in tmp directory, in commit operation we should move the file to target path. Do I understand that right? @chessplay

Reply:
@TyrantLucifer yes,you are right

Reply:
@CalvinKirs because the rename method is   mainly  used to move a  file ,so we should create the target diretory first. The createFile method could create target directory recursively and create a empty file